src/debug/nodejsDebugProvider.ts

```diff
@@ -0,0 +1,99 @@
+import * as path from "path";
+import * as vscode from "vscode";
+import { ChildProcess } from "child_process";
+
+import { IDebugProvider, PortInfo } from "./debugProvider";
+import { suggestedShellForContainer } from './debugUtils';
+import * as config from '../components/config/config';
+import { Kubectl } from "../kubectl";
+import { IDockerfile } from "../docker/parser";
+import { kubeChannel } from "../kubeChannel";
+
+interface ExecResult {
+    readonly proxyProcess: ChildProcess;
+}
+const debuggerType = 'nodejs';
+
+export class NodejsDebugProvider implements IDebugProvider {
+    remoteRoot: string;
+    shell: string;
+
+    public getDebuggerType(): string {
+        return debuggerType;
+    }
+
+    public async isDebuggerInstalled(): Promise<boolean> {
+        return true;
+    }
+
+    public async startDebugging(workspaceFolder: string, sessionName: string, port: number): Promise<boolean> {
+
+        const debugConfiguration: vscode.DebugConfiguration = {
+            type: "node",
+            request: "attach",
+            name: sessionName,
+            hostName: "localhost",
+            skipFiles: [
+                "<node_internals>/**/*.js"
+            ],
+            port
+        };
+        if (this.remoteRoot) {
+            debugConfiguration['remoteRoot'] = this.remoteRoot;
+        }
+        const currentFolder = vscode.workspace.workspaceFolders.find((folder) => folder.name === path.basename(workspaceFolder));
+        return await vscode.debug.startDebugging(currentFolder, debugConfiguration);
+    }
+
+    isSupportedImage(baseImage: string): boolean {
+        return false;
+    }
+
+    resolvePortsFromFile(dockerfile: IDockerfile, env: {}): Promise<PortInfo> {
+        return undefined;
+    }
+
+    public async resolvePortsFromContainer(kubectl: Kubectl, pod: string, podNamespace: string, container: string): Promise<PortInfo> {
+        this.shell = await suggestedShellForContainer(kubectl, pod, podNamespace, container);
+        const inspectModeResult = await this.setNodeInspectMode(kubectl, pod, podNamespace, container);
+        if (!inspectModeResult) {
+            kubeChannel.showOutput('Unable to set Node.js in the container to inspect mode.');
+            return undefined;
+        }
+        this.remoteRoot = await this.tryGetRemoteRoot(kubectl, pod, podNamespace, container);
+        const rawDebugPortInfo = config.getNodejsDebugPort() || '9229';
```

Comment:
Looks like the function now returns a number but you are `||`-ing it with a string.  Should it be:
```suggestion
        const rawDebugPortInfo = config.getNodejsDebugPort() || 9229;
```